x/mongo/driver/mongocrypt/mongocrypt.go

```diff
@@ -410,7 +414,51 @@ func (m *MongoCrypt) createErrorFromStatus() error {
 	return errorFromStatus(status)
 }
 
+// needsKmsProvider returns true if provider was initially set to an empty document.
+// An empty document signals the driver to fetch credentials.
+func needsKmsProvider(kmsProviders bsoncore.Document, provider string) bool {
+	val, err := kmsProviders.LookupErr(provider)
+	if err != nil {
+		// KMS provider is not configured.
+		return false
```

Comment:
Is it always okay to ignore the error here and return false? Will a user never need the error?

Reply:
It seems safe to ignore the error as either `kmsProviders` does not contain `provider` or `kmsProviders` is malformatted. And since `kmsProviders` is not a direct user input, we are pretty much eating our own dog food.